examples/EncodeDecode/main.roc

```diff
@@ -1,125 +1,137 @@
-app [main!] {
-    cli: platform "https://github.com/roc-lang/basic-cli/releases/download/0.20.0/X73hGh05nNTkDHU06FHC0YfFaQB1pimX7gncRcao5mU.tar.br",
-    json: "https://github.com/lukewilliamboswell/roc-json/releases/download/0.13.0/RqendgZw5e1RsQa3kFhgtnMP8efWoqGRsAvubx4-zus.tar.br",
+app [main!] { pf: platform "../platform/main.roc" }
+
+import pf.Stdout
+
+# Example demonstrating the new Encode module with static dispatch
+# Based on: https://github.com/roc-lang/roc/commit/22cf61ff9332f0de7a0d5d7f42b7f5836232a744
+#
+# The Encode module uses static dispatch via where clauses:
+# - Str.encode requires: where [fmt.encode_str : fmt, Str -> List(U8)]
+# - List.encode requires: where [fmt.encode_list : fmt, List(item), (item, fmt -> List(U8)) -> List(U8)]
+
+# Define a custom JSON-like format type with the required methods
+JsonFormat := [Format].{
+    # Method required by Str.encode where clause
+    encode_str : JsonFormat, Str -> List(U8)
+    encode_str = |_fmt, str| {
+        # Wrap string in quotes and convert to bytes
+        quoted = "\"${str}\""
+        Str.to_utf8(quoted)
+    }
+
+    # Method required by List.encode where clause  
+    encode_list : JsonFormat, List(item), (item, JsonFormat -> List(U8)) -> List(U8)
+    encode_list = |fmt, items, encode_item| {
+        var $result = ['[']
+        var $first = Bool.True
+        
+        for item in items {
+            if $first {
+                $first = Bool.False
+            } else {
+                $result = $result.append(',')
+            }
+            encoded_item = encode_item(item, fmt)
+            $result = $result.concat(encoded_item)
+        }
+        
+        $result.append(']')
+    }
 }
 
-import json.Json
-import cli.Stdout
-import cli.Arg exposing [Arg]
-### start snippet impl
+# Define a Person type that can be encoded as a JSON object
+Person := [Person({ name : Str, age : U64 })].{
+    # Custom encode method for Person - encodes as JSON object
+    encode : Person, JsonFormat -> List(U8)
+    encode = |self, fmt| {
+        # Get the inner record via pattern match
+        match self {
+            Person({ name, age }) => {
+                # Encode name as JSON string
+                name_bytes = name.encode(fmt)
+                
+                # Encode age as number (no quotes)
+                age_bytes = Str.to_utf8(age.to_str())
+                
+                # Build: {"name":"...","age":...}
+                var $result = Str.to_utf8("{\"name\":")
+                $result = $result.concat(name_bytes)
+                $result = $result.concat(Str.to_utf8(",\"age\":"))
+                $result = $result.concat(age_bytes)
+                $result = $result.concat(Str.to_utf8("}"))
+                $result
+            }
+        }
+    }
+}
```

Comment:
I believe this will be more auto derived in the future, can you add a TODO comment above this to make this use auto derive when it is implemented?